crates/storage/src/connection/block.rs

```diff
@@ -443,6 +443,27 @@ pub(super) fn first_block_without_receipts(
     }
 }
 
+pub(super) fn highest_block_with_all_class_definitions_downloaded(
```

Comment:
Do you have a decent feeling for how well/poorly this scales?

Reply:
goerli@925k blocks: 50ms on my machine (m2ssd, ryzen3700x, 32g ram)

Reply:
Sounds fine to me then